esileapclient/common/utils.py

```diff
@@ -0,0 +1,88 @@
+import re
+import operator
+import logging
+
+# Configure the logger
+LOG = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+class PropertyFilterParser:
+    OPS = {
+        '>=': operator.ge,
+        '<=': operator.le,
+        '>': operator.gt,
+        '<': operator.lt,
+        '=': operator.eq,
+    }
+
+    def __init__(self):
+        self.operator_pattern = self.build_operator_pattern()
+        pattern = (
+            rf'([^><=]+)'
+            rf'({self.operator_pattern})'
+            rf'(.+)'
+        )
+        self.filter_pattern = re.compile(pattern)
+
+    def build_operator_pattern(self):
+        """Build a regex pattern for the operators in OPS."""
+        return '|'.join(re.escape(op) for op in self.OPS.keys())
+
+    def convert_value(self, value_str):
+        """Convert a value string to an appropriate type for comparison."""
+        try:
+            return int(value_str)
+        except ValueError:
+            try:
+                return float(value_str)
+            except ValueError:
+                return value_str
+
+    def parse_property_filter(self, filter_str):
+        """Parse a property filter string into a key, operator, and value."""
+        match = self.filter_pattern.match(filter_str)
+        if not match:
+            raise ValueError(f"Invalid property filter format: {filter_str}")
+        key, op_str, value_str = match.groups()
+        if op_str not in self.OPS:
+            raise ValueError(f"Invalid operator in property filter: {op_str}")
+        value = self.convert_value(value_str)
+        return key.strip(), self.OPS[op_str], value
+
+
+parser = PropertyFilterParser()
+
+
+def node_matches_property_filters(node, property_filters):
+    """Check if a node matches all property filters."""
+    for key, op, value in property_filters:
+        if key not in node['properties']:
+            return False
+        node_value = parser.convert_value(node['properties'].get(key, ''))
+        if not op(node_value, value):
+            return False
+    return True
+
+
+def filter_nodes_by_properties(nodes, properties):
+    """Filter a list of nodes based on property filters."""
+    if not properties:
+        return nodes
+    property_filters = []
+    for prop in properties:
+        try:
+            property_filters.append(parser.parse_property_filter(prop))
+        except ValueError as e:
+            LOG.error(f"Error parsing property filter '{prop}': {e}")
+            raise  # Raise the exception to stop execution
+
+    filtered_nodes = [
+        node for node in nodes
+        if node_matches_property_filters(node, property_filters)
+    ]
+
+    if not filtered_nodes:
+        LOG.error("No nodes match the specified properties.")
```

Comment:
I don't think you need this; returning an empty list if no nodes match isn't an error, and the handling of an empty list should be done within the `node list` code. I think it's fine for all of that to just return... nothing; that's what the other OpenStack clients do.

Reply:
okay.